### DISCUSSION TITLE
The Legend of the Simple Table

### DISCUSSION THREAD

**Discussion by james-rae** [fgpv-vpgf/fgpv-vpgf#4112] [Team Posts]
## In the Beginning...

The initial path of the v3 migration to plugins had the concept of RAMP having a default HTML table with no features (the "Simple Table"). The fancy table would be loaded as a plugin if a map author desired it.  The main rationale was this would keep the significant size of `ag-grid` library out of `rv-main.js`.

## Refactor Adventures

As work on conversion grew in scope, the Simple Table became de-prioritized (as it does not exist in v2, there was no real obligation to add it to v3).  A version of the Simple Table did exist, but had a number of bugs.  It was decided for v3.0.0 that we would auto-bundle the fancy grid (the argument being most authors would want it anyway), and leave the original vision to be implemented or discarded at a later date.

## Useful Links

In the case that we do decide to continue with the Simple Table, the following things can be used to steal previous efforts.

[Documentation](http://fgpv-vpgf.github.io/fgpv-vpgf/v3.0.0-b2/#/developer/datatables) on the behavior of Simple Table, and differences compared to Fancy Table.

The [version](https://github.com/fgpv-vpgf/fgpv-vpgf/tree/v3.0.0-b1) that has simple table in use by default. [Samples](http://fgpv.cloudapp.net/demo/v3.0.0-b1/prod/samples/) of that version.

**Comment by mweech** [comment]
Here are my main concerns with the "Simple Table" based on the documentation in the link you provided.  The following is a list of features that are NOT supported on the Simple Table, each of which I think we need to address.

- Customize through config: Why is this not doable for a simple table?  Would this not just customize which dataset attributes get added to a table?  
- Feature Details Shortcut: Less important with crosshairs based identify available.  However, if a table is customized to hide some columns (as noted above), we would need to be able to view hidden datagrid attributes in the Feature Details shortcut.  Why is a reduced feature table mean we can't include such a JavaScript shortcut on the table rows?
- Zoom Shortcut: Less important with crosshairs based identify available, but also means that the datagrid is pretty useless geographically since there is no way to tell what feature that row specifically matches.  Why is this not doable without a heavy weight datagrid library??
- Custom Title: Why not? Why do we need a fancy datagrid to change the table title?
- Mobile Compatible: This is a must.  Why is this not doable..? Wouldn't the panels API handle this?
- Synced with Layer Visibility: Again, why not?
- Filter by Map Extent: Same as above...

The simple table was supposed to be less heavy weight of a library, not have no features at all.
